File: Python/MinTimeVisitingAllPoints.py
# https://leetcode.com/problems/minimum-time-visiting-all-points/description/?envType=daily-question&envId=2026-01-12

from typing import List

class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        ans = 0
        for i in range(len(points) - 1):
            curr_x, curr_y = points[i]
            target_x, target_y = points[i + 1]
            ans += max(abs(target_x - curr_x), abs(target_y - curr_y))

        return ans
    
if __name__ == "__main__":
    points = [[1,1],[3,4],[-1,0]]
    print(Solution().minTimeToVisitAllPoints(points))

File: MinTimeVisitingAllPoints.java
// https://leetcode.com/problems/minimum-time-visiting-all-points/description/?envType=daily-question&envId=2026-01-12

class MinTimeVisitingAllPoints {
    public int minTimeToVisitAllPoints(int[][] points) {
        int ans = 0;
        for(int i = 0; i < points.length - 1; i++){
            int currX = points[i][0];
            int currY = points[i][1];
            int targetX = points[i + 1][0];
            int targetY = points[i + 1][1];
            ans += Math.max(Math.abs(targetX - currX), Math.abs(targetY - currY));
        }
        return ans;
    }

    public static void main(String[] args) {
        MinTimeVisitingAllPoints solution = new MinTimeVisitingAllPoints();
        int[][] points = {{1,1},{3,4},{6,1}};
        int result = solution.minTimeToVisitAllPoints(points);
        System.out.println("Minimum time to visit all points: " + result);
    }
}